dispatcher/dispatcher.go

```diff
@@ -274,41 +300,81 @@ func (d *IotxDispatcher) dispatchAction(ctx context.Context, chainID uint32, msg
 	if atomic.LoadInt32(&d.shutdown) != 0 {
 		return
 	}
-	d.enqueueEvent(&actionMsg{
-		ctx:     ctx,
-		chainID: chainID,
-		action:  (msg).(*iotextypes.Action),
-	})
+	subscriber := d.subscriber(chainID)
+	if subscriber == nil {
+		log.L().Debug("no subscriber for this chain id, drop the action", zap.Uint32("chain id", chainID))
+		return
+	}
+	d.actionChanLock.Lock()
+	defer d.actionChanLock.Unlock()
```

Comment:
this is not necessary?
only `HandlBroadcast()` calls `dispatchAction` that will send to this channel